crates/sats/src/typespace.rs

```diff
@@ -232,14 +232,65 @@ pub trait GroundSpacetimeType {
     fn get_type() -> AlgebraicType;
 }
 
-/// A trait for Rust types that can be represented as an [`AlgebraicType`]
-/// provided a typing context `typespace`.
+/// This trait makes types self-describing, allowing them to automatically register their structure
+/// with SpacetimeDB. This is used to tell SpacetimeDB about the structure of a module's tables and
+/// reducers.
+///
+/// Deriving this trait also derives [`Serialize`](crate::ser::Serialize), [`Deserialize`](crate::de::Deserialize),
+/// and [`Debug`](std::fmt::Debug). (There are currently no trait bounds on `SpacetimeType` documenting this fact.)
+/// These traits are used to convert Rust data structures to other formats, suitable for storing on disk or passing over the network. `Debug` is simply for debugging convenience.
+///
+/// Any Rust type implementing `SpacetimeType` can be used in table and reducer declarations. A derive macro is provided, and can be used on both structs and enums:
```

Comment:
Can you be more specific? I believe we don't allow tuple structs. I'm not sure whether or not we allow unit structs. I think on enums we only allow unit variants and single-element tuple variants, not struct variants or n-tuple variants. I could be wrong though; you should check.

Reply:
Should have said "Any Rust type implementing `SpacetimeType` can be used as a table column or reducer argument."